### PR TITLE
feat: #13 extend profile for VaccinationScheme

### LINKED ISSUE #13: Model VaccinationScheme

### PR DESCRIPTION
**Pull request by n-krueger**
Closes #13

### CODE REVIEW

File: ig/input/fsh/VaccinationScheme.fsh [hunk 1]
```diff
@@ -6,9 +6,56 @@ Description: """
...
+
+* extension[isPreferred]
+  * value[x] 1..1 MS
+  * value[x] only boolean
+
+* extension[ageStart]
+  * value[x] 1..1 MS
+  * value[x] only Quantity
+  * value[x] obeys age-1
+  * value[x].code from http://hl7.org/fhir/ValueSet/all-time-units
```
Comment (tklausen) [inline ig/input/fsh/VaccinationScheme.fsh]: I think we should forbid any extensions to the Quantity element.

```suggestion
  * value[x]
    * code from http://hl7.org/fhir/ValueSet/all-time-units
    * extension 0..0
```

File: ig/input/fsh/VaccinationScheme.fsh [hunk 2]
```diff
@@ -6,9 +6,56 @@ Description: """
...
+
+* extension[ageEnd]
+  * value[x] 1..1 MS
+  * value[x] only Quantity
+  * value[x] obeys age-1
+  * value[x].code from http://hl7.org/fhir/ValueSet/all-time-units
```
Comment (tklausen) [inline ig/input/fsh/VaccinationScheme.fsh]: I think we should forbid any extensions to the Quantity element.

```suggestion
  * value[x]
    * code from http://hl7.org/fhir/ValueSet/all-time-units
    * extension 0..0
```

File: ig/input/fsh/VaccinationScheme.fsh (hunk 1 above)
Comment (tklausen) [inline ig/input/fsh/VaccinationScheme.fsh]: Why did you use that huge ValueSet [http://hl7.org/fhir/ValueSet/all-time-units](http://hl7.org/fhir/ValueSet/all-time-units) to specify an age?
Can we instead use [http://hl7.org/fhir/ValueSet/age-units](http://hl7.org/fhir/ValueSet/age-units)?

The same applies to ageEnd.

Comment (tklausen) [inline ig/input/fsh/VaccinationScheme.fsh]: I think we should set value, system, code (and maybe even unit) of value[x] to `1..1 MS`

The same applies to ageEnd.

File: ig/input/fsh/VaccinationScheme.fsh [hunk 3]
```diff
@@ -6,9 +6,56 @@ Description: """
   VaccinationScheme models a process consisting of immunizations through wich a patient can reach immunity against a target
   disease.
 """
```
Comment (tklausen) [inline ig/input/fsh/VaccinationScheme.fsh:8]: Can we add the MS flag to `code` so it shows up in the must-support snapshot table?

```suggestion
"""
* code 1..1 MS
```

File: ig/input/fsh/VaccinationScheme.fsh [hunk 4]
```diff
@@ -6,9 +6,56 @@ Description: """
...
 Extension: VPVaccinationSchemeExtension
 Id: vp-vaccination-scheme-extension
 Title: "VaccinationSchemeExtension"
 Description: ""
+* extension contains
+  name 1..1 MS and
+  type 1..1 MS and
+  isPreferred 1..1 MS and
+  ageStart 0..1 MS and
+  ageEnd 0..1 MS
```
Comment (tklausen) [inline ig/input/fsh/VaccinationScheme.fsh]: There should be a mandatory reference to VPMedication.

**Review by tklausen — CHANGES_REQUESTED** [review]
Nice job with that complex extension and the newly defined CodeSystem/ValueSystem. However, I have a few small suggestions:

File: ig/input/fsh/VaccinationScheme.fsh (hunk 1 above)
Comment (n-krueger) [reply ig/input/fsh/VaccinationScheme.fsh]: WONTFIX

File: ig/input/fsh/VaccinationScheme.fsh (hunk 2 above)
Comment (n-krueger) [reply ig/input/fsh/VaccinationScheme.fsh]: WONTFIX

File: ig/input/fsh/VaccinationScheme.fsh (hunk 1 above)
Comment (n-krueger) [reply ig/input/fsh/VaccinationScheme.fsh]: Addressed at https://github.com/FHIR-Vaccination-Pass/fhir-implementation-guide/pull/27/commits/5594936ff7129c275bc2504dd5d2d616b9cd0ff2

Comment (n-krueger) [reply ig/input/fsh/VaccinationScheme.fsh]: Addressed at f6edd73d

File: ig/input/fsh/VaccinationScheme.fsh (hunk 4 above)
Comment (n-krueger) [reply ig/input/fsh/VaccinationScheme.fsh]: Addressed at 2f3e1a4

**Review by tklausen — APPROVED** [review]
Nice work!

**Pull request merged**